### ISSUE TITLE
Snapshot UI test 

### ISSUE THREAD

**Issue by alltheseas** [damus-io/damus#1562]
> Snapshot tests are great for detecting aesthetic or layout issues (and they can be easy to implement).

https://github.com/damus-io/damus/issues/1364#issuecomment-1736420718

**Comment by alltheseas** [comment]
@danieldaquino recalls some previews are crashing. Nostrdb allocate diskspace -> crashes preview. Will noticed crashes CI. Migrating in-memory events to profiles need to make sure database is initialized. 

Tangential: xcode cloud allows cloud CI. Damus CI tests are broken.

**Comment by danieldaquino** [comment]
I will give this one a try now.

If I can implement this, it might be useful for some of the other sprint items, as I will be able to setup snapshot tests for them.

**Comment by danieldaquino** [comment]
@jb55, the test target is currently failing to build _(which blocks this ticket)_, so I will see if I can quickly come up with a fix for that as well. _(I don't know if there is a ticket for this already, please let me know if there is a ticket I can link against)_

**Comment by danieldaquino** [comment]
I tried to adapt the tests, but it seems that adapting `UserCacheManagerTests` is non-trivial, so I sent a patch to temporarily disable those tests, and I opened a separate ticket to track fixing those tests: #1586

**Comment by danieldaquino** [comment]
@jb55, @alltheseas, sent a patch with the setup of snapshot testing, and created one example snapshot test: https://groups.google.com/a/damus.io/g/patches/c/Tkc1C-Ud9dE

Here is roughly how it works:

1. I wrote the test, and the test calls a function called `assertSnapshot`
2. When I run it the first time, it will automatically take a snapshot of the view under test, save it to the repo, and fail _(to catch my attention to review the image)_ Here the snapshot from this step (I am checking whether the text editor will wrap text):
![testTextWrapperViewWillWrapText 1](https://github.com/damus-io/damus/assets/24692108/503f182f-ae75-4296-be38-b60419db0c7a)
3. If I am happy, I just need to commit the image along with other code changes.
4. When I run the test again, it passes (It takes a snapshot and compares with the baseline, which in this case are the same)
5. If I change the code in a way that breaks this layout, and I run the test again, it will fail and show me the two snapshots (the baseline and the current one)
6. If the current snapshot is expected (e.g. An intentional layout change), I can replace the baseline
7. If the current snapshot is not expected, then we have detected a bug.
8. When baseline snapshot change, they get committed to the repo, which has the added benefit of giving the reviewer a change to visually see how the UI changes as a result of the code changes.
